src/lib/dai.c

```diff
@@ -149,15 +149,16 @@ const struct device *zephyr_dev[] = {
 
 static const struct device *dai_get_zephyr_device(uint32_t type, uint32_t index)
 {
-	const struct dai_config *cfg;
+	struct dai_config cfg;
 	int dir;
 	int i;
 
 	dir = (type == SOF_DAI_INTEL_DMIC) ? DAI_DIR_RX : DAI_DIR_BOTH;
 
 	for (i = 0; i < ARRAY_SIZE(zephyr_dev); i++) {
-		cfg = dai_config_get(zephyr_dev[i], dir);
-		if (cfg && cfg->type == type && cfg->dai_index == index)
+		if (dai_config_get(zephyr_dev[i], &cfg, dir))
+			return NULL;
```

Comment:
@juimonen is this correct? Should you continue here instead?

Reply:
I mean dai_config_get should return 0 on success, otherwise negative. You kind of always get some config back, it is just testing the arguments you give are not bogus... so in that sense this can't fail, just added the check for completeness.

Reply:
even if this is not possible in practice, the right thing to do here is continue instead of return NULL right?

Reply:
yes you're right.